asrtoolkit/data_handlers/json.py

```diff
@@ -85,7 +86,7 @@ def assign_if_present(
         assign_if_present("corrected_transcript", "text")
         assign_if_present("formatted_transcript", "formatted_text")
         assign_if_present("punctuated_transcript", "formatted_text")
-        assign_if_present("speakerInfo", "speaker", "ID")
+        assign_if_present("speakerInfo", "speaker", proc_val=sanitize)
```

Comment:
This not only pulls the correct value from the JSON, but it also eliminates whitespace in the speaker id, which is necessary for many output formats.

Reply:
This may have a bug for files without speaker information as we saw when pairing today.

Reply:
I've confirmed this line doesn't assign speaker to `UnknownSpeaker` when `ID` is undefined. This creates an invalid STM file for `convert_transcript`.

This fixed the error for me locally.
```suggestion
        assign_if_present("speakerInfo", "speaker", "ID", proc_val=sanitize)
```